src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1388,9 +1392,18 @@ private function dumpValue($value, $interpolate = true)
      * @param string $class
      *
      * @return string
+     *
+     * @throws RuntimeException
      */
     private function dumpLiteralClass($class)
     {
+        if (false !== strpos($class, '$')) {
+            throw new RuntimeException('Cannot dump definitions which have a variable class name.');
+        }
+        if (0 !== strpos($class, "'") || !preg_match('/^\'[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*(\\\{2}[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)*\'$/', $class)) {
+            throw new RuntimeException('Cannot dump definition because of invalid class name');
```

Comment:
the invalid class name should be included in the exception message to make it easier to debug